### ISSUE TITLE
Decreased accuracy after conversion to TensorRT

### ISSUE THREAD

**Issue by JimKlingshirn** [NVIDIA/retinanet-examples#34]
I derived a c++ app (attached) from cppapi/infer.cpp for evaluating a TensorRT model on the COCO validation set. When I convert your pretrained retinanet_rn50fpn.pth and evaluate it on a TX2, it's accuracy is reduced.  I'm expecting  IoU=0.50:0.95 score of 0.358.  Instead, I measure 0.277.    

I'm using an unmodified version of this repository. According to dpkg, I'm using tensorrt                                   5.0.6.3-1+cuda10.0 on the TX2.

Conversion, evaluation parameters:
```
retinanet export /coco/retinanet_rn50fpn.pth --opset=9 --batch=1 --size=800 --verbose /coco/retinanet_rn50fpn_800.onnx
./export retinanet_rn50fpn_800.onnx retinanet_rn50fpn_800.plan
./infer_coco_val retinanet_rn50fpn_800.plan /mnt/data/datasets/coco_2017/list.txt detections_800.json
```
Thanks for the excellent repository, and thanks in advance for any ideas. 

[infer_coco_val.zip](https://github.com/NVIDIA/retinanet-examples/files/3283233/infer_coco_val.zip)

**Comment by JimKlingshirn** [comment]
Looking at some of the TX2 detection outputs, the classes look good, but sometimes the bounding boxes are shifted towards the top, left.  Possibly a problem with the anchor box decoder, or the box offsets from here? :  [extras/cppapi/export.cpp](https://github.com/NVIDIA/retinanet-examples/blob/master/extras/cppapi/export.cpp#L47)

![1584](https://user-images.githubusercontent.com/5315006/59457828-08af5b00-8dce-11e9-9c14-8f9a66de9125.png)
 
![7784](https://user-images.githubusercontent.com/5315006/59457808-0220e380-8dce-11e9-9943-550e1c089125.png)

**Comment by pkashinkunti** [comment]
Hi JimKlingshirn,

There's a few things I noticed in your issue description:

* By using `--size 800` in your `retinanet export` cmd, you are specifying that the TRT engine produced should have a fixed input of 800x800. This input size is smaller than what was used to calculate the IoU scores (and smaller than what the model was trained with) so I would expect the accuracy to be reduced.

* In general, the cppapi code is more intended as a minimal example to demonstrate the C++ workflow, not necessarily to demonstrate accuracy.

* Looking through the cpp file used to evaluate accuracy, the image preprocessing used is different than what is used during `retinanet infer`. It is resizing the images to match the input dimensions of the TRT engine. I recommend you check out how TRT inference is done in the rest of the codebase (https://github.com/NVIDIA/retinanet-examples/blob/master/csrc/extensions.cpp#L100 and https://github.com/NVIDIA/retinanet-examples/blob/master/retinanet/data.py#L47).
